source/priority_queue.c

```diff
@@ -159,6 +159,10 @@ int aws_priority_queue_init_dynamic(
     size_t item_size,
     aws_priority_queue_compare_fn *pred) {
 
+    if (queue == NULL) {
```

Comment:
You can use AWS_ERROR_PRECONDITION here soon (as soon as #434 is merged)

Reply:
Good idea! I've gone ahead and made those changes.